### DISCUSSION TITLE
Grid access through the model server

### DISCUSSION THREAD

**Discussion by chrisdaviesweb** [ExpressionEngine/ExpressionEngine#2985] [Feature Requests]
Access to grid field data through the 'ChannelEntry' model service.

**Comment by intoeetive** [comment]
@chrisdaviesweb would you be willing to elaborate on what you have in mind here?

**Comment by rowansentesy** [comment]
I'm not sure if my current interest is related to the OP's post above, but it is the most similar so I'll start here.

I would like to see the Grid functionality exposed via Model Services. I have found working within the model services for most things to be awesome and very helpful for a lot of addon development work, however I ran into some snags when trying to generate a new Grid field, and set its settings. I was unable to do so with Model services. 

I setup a function in an addon updater to create this new Grid field, but had to do some manual SQL queries to get it to generate everything properly.

- insert into exp_grid_columns
- create table exp_channel_grid_field_*
- Update exp_channel_fields, set field_settings for new grid field

These above steps I was able to do using ee()->db->query functions. 

It would be great if I could use `$field = ee('Model')->make('ChannelField')` and
`$field->grid_columns = [..cols..];`
`$field->save()`
etc.

Perhaps what the OP is referring to is being able to access grid data from the `ee('Model')->get('ChannelEntry')...` call. 

I was able to work around this by using the `get_entry_rows` function call
```
// grid_model.php line 349:
get_entry_rows($entry_ids, $field_id, $content_type, $options = array(), $reset_cache = FALSE, $fluid_field_data_id = 0)
```

I used this in my code like this:
```
ee()->load->model('grid_model');
                $rows = ee()->grid_model->get_entry_rows(
                    $entry_id,
                    $grid_field_id,
                    'channel',
                    ['sort' => 'ASC', 'orderby' => 'row_order'],
                    FALSE
                );
```

It would be nice, if this was an accessible object through the ChannelEntry model. Maybe using a `->with('Grid')` or something.

Let me know if you'd like me to post my enhancement request elsewhere, or if this is a good spot for now.

thank you!
Rowan
